shim/integration/kvm_get_fpu.cpp

```diff
@@ -0,0 +1,117 @@
+/// @copyright
+/// Copyright (C) 2020 Assured Information Security, Inc.
+///
+/// @copyright
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+///
+/// @copyright
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+///
+/// @copyright
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+/// SOFTWARE.
+
+#include <integration_utils.hpp>
+#include <ioctl.hpp>
+#include <kvm_fpu.hpp>
+#include <shim_platform_interface.hpp>
+
+#include <bsl/array.hpp>
+#include <bsl/convert.hpp>
+#include <bsl/enable_color.hpp>
+#include <bsl/exit_code.hpp>
+#include <bsl/safe_idx.hpp>
+#include <bsl/safe_integral.hpp>
+
+namespace
+{
+    /// @brief defines the Expected for registers
+    constexpr auto EXPECTED{0x12345678_u32};
+    /// @brief defines the VAL for registers
+    constexpr auto MYSIZE_FR{128_u64};
+    /// @brief defines the VAL for registers
+    constexpr auto MYSIZE_REG{32_u64};
+    /// @brief defines the VAL for registers
```

Comment:
These comments are repeated. Though it's obvious from the names of the variables, could you fix up the comments to indicate what the registers are?